src/attesters/api/enclave_attester_register.c

```diff
@@ -28,6 +28,16 @@ enclave_attester_err_t enclave_attester_register(const enclave_attester_opts_t *
 		}
 	}
 
+	if (opts->flags & ENCLAVE_ATTESTER_OPTS_FLAGS_SNP_GUEST) {
+		if (!is_snpguest_supported()) {
+			// clang-format off
+			RTLS_DEBUG("failed to register the attester '%s' due to lack of SNP Guest capability\n",
```

Comment:
If user don't specify the --debug as start  parameter, the failure will be hidden. 
I think here we should use RTLS_ERR to force output the info and notify user the failure root cause.

Reply:
Whether to use RTLS_ERR or RTLS_DEBUG was discussed carefully in enclave-tls.
The reason is as followed:
If an offline user uses a shelter to verify the sev/tdx certificate on a non-sev/tdx platform, the registration of the sev/tdx attester instance must fail, but it does not mean that it is wrong.

In addition, please see the code of tdx guest regidter.
```c
        if (opts->flags & ENCLAVE_ATTESTER_OPTS_FLAGS_TDX_GUEST) {
                if (!is_tdguest_supported()) {
                        // clang-format off
                        RTLS_DEBUG("failed to register the attester '%s' due to lack of TDX Guest capability\n",
                                   opts->type);
                        // clang-format on
                        return -ENCLAVE_ATTESTER_ERR_CPU_UNSUPPORTED;
                }
        }

```